Prevent conditional expression between different enum types using `static_cast`

I get the following error when compiling with `-std=gnu++20`:

```
src/raygui.h:2089:41: warning: conditional expression between different enumeration types ('GuiControlProperty' and 'GuiDefaultProperty') is deprecated [-Wdeprecated-enum-compare-conditional]
                                        ? BORDER_COLOR_DISABLED
                                        ^ ~~~~~~~~~~~~~~~~~~~~~
src/raygui.h:2096:41: warning: conditional expression between different enumeration types ('GuiControlProperty' and 'GuiDefaultProperty') is deprecated [-Wdeprecated-enum-compare-conditional]
                                        ? BORDER_COLOR_DISABLED
                                        ^ ~~~~~~~~~~~~~~~~~~~~~
src/raygui.h:2103:41: warning: conditional expression between different enumeration types ('GuiControlProperty' and 'GuiDefaultProperty') is deprecated [-Wdeprecated-enum-compare-conditional]
                                        ? BORDER_COLOR_DISABLED
                                        ^ ~~~~~~~~~~~~~~~~~~~~~
src/raygui.h:2129:42: warning: conditional expression between different enumeration types ('GuiControlProperty' and 'GuiDefaultProperty') is deprecated [-Wdeprecated-enum-compare-conditional]
      DEFAULT, (state == STATE_DISABLED) ? BORDER_COLOR_DISABLED : LINE_COLOR));
                                         ^ ~~~~~~~~~~~~~~~~~~~~~   ~~~~~~~~~~
src/raygui.h:2194:54: warning: conditional expression between different enumeration types ('GuiControlProperty' and 'GuiDefaultProperty') is deprecated [-Wdeprecated-enum-compare-conditional]
                                                     ? BORDER_COLOR_DISABLED
                                                     ^ ~~~~~~~~~~~~~~~~~~~~~
src/raygui.h:2197:54: warning: conditional expression between different enumeration types ('GuiControlProperty' and 'GuiDefaultProperty') is deprecated [-Wdeprecated-enum-compare-conditional]
                                                     ? BASE_COLOR_DISABLED
                                                     ^ ~~~~~~~~~~~~~~~~~~~
```

---

Since conditional expressions between different enum types are now deprecated in C++20, perhaps `static_cast` should be used to cast one enum type into the other one first?

**Note:** Please disregard the line numbers as I am using an outdated verson of `raygui` (which I may have accidentally formatted). In the latest version on the `master` branch (commit 6aae838), the line numbers should be 1642 and onwards.

--- a/src/raygui.h
+++ b/src/raygui.h
@@ -1639,9 +1639,9 @@ int GuiGroupBox(Rectangle bounds, const char *text)
 
     // Draw control
     //--------------------------------------------------------------------
-    GuiDrawRectangle(RAYGUI_CLITERAL(Rectangle){ bounds.x, bounds.y, RAYGUI_GROUPBOX_LINE_THICK, bounds.height }, 0, BLANK, GetColor(GuiGetStyle(DEFAULT, (state == STATE_DISABLED)? BORDER_COLOR_DISABLED : LINE_COLOR)));
-    GuiDrawRectangle(RAYGUI_CLITERAL(Rectangle){ bounds.x, bounds.y + bounds.height - 1, bounds.width, RAYGUI_GROUPBOX_LINE_THICK }, 0, BLANK, GetColor(GuiGetStyle(DEFAULT, (state == STATE_DISABLED)? BORDER_COLOR_DISABLED : LINE_COLOR)));
-    GuiDrawRectangle(RAYGUI_CLITERAL(Rectangle){ bounds.x + bounds.width - 1, bounds.y, RAYGUI_GROUPBOX_LINE_THICK, bounds.height }, 0, BLANK, GetColor(GuiGetStyle(DEFAULT, (state == STATE_DISABLED)? BORDER_COLOR_DISABLED : LINE_COLOR)));
+    GuiDrawRectangle(RAYGUI_CLITERAL(Rectangle){ bounds.x, bounds.y, RAYGUI_GROUPBOX_LINE_THICK, bounds.height }, 0, BLANK, GetColor(GuiGetStyle(DEFAULT, (state == STATE_DISABLED)? (int)BORDER_COLOR_DISABLED : (int)LINE_COLOR)));
+    GuiDrawRectangle(RAYGUI_CLITERAL(Rectangle){ bounds.x, bounds.y + bounds.height - 1, bounds.width, RAYGUI_GROUPBOX_LINE_THICK }, 0, BLANK, GetColor(GuiGetStyle(DEFAULT, (state == STATE_DISABLED)? (int)BORDER_COLOR_DISABLED : (int)LINE_COLOR)));
+    GuiDrawRectangle(RAYGUI_CLITERAL(Rectangle){ bounds.x + bounds.width - 1, bounds.y, RAYGUI_GROUPBOX_LINE_THICK, bounds.height }, 0, BLANK, GetColor(GuiGetStyle(DEFAULT, (state == STATE_DISABLED)? (int)BORDER_COLOR_DISABLED : (int)LINE_COLOR)));
 
     GuiLine(RAYGUI_CLITERAL(Rectangle){ bounds.x, bounds.y - GuiGetStyle(DEFAULT, TEXT_SIZE)/2, bounds.width, (float)GuiGetStyle(DEFAULT, TEXT_SIZE) }, text);
     //--------------------------------------------------------------------
@@ -1662,7 +1662,7 @@ int GuiLine(Rectangle bounds, const char *text)
     int result = 0;
     GuiState state = guiState;
 
-    Color color = GetColor(GuiGetStyle(DEFAULT, (state == STATE_DISABLED)? BORDER_COLOR_DISABLED : LINE_COLOR));
+    Color color = GetColor(GuiGetStyle(DEFAULT, (state == STATE_DISABLED)? (int)BORDER_COLOR_DISABLED : (int)LINE_COLOR));
 
     // Draw control
     //--------------------------------------------------------------------
@@ -1710,7 +1710,7 @@ int GuiPanel(Rectangle bounds, const char *text)
     //--------------------------------------------------------------------
     if (text != NULL) GuiStatusBar(statusBar, text);  // Draw panel header as status bar
 
-    GuiDrawRectangle(bounds, RAYGUI_PANEL_BORDER_WIDTH, GetColor(GuiGetStyle(DEFAULT, (state == STATE_DISABLED)? BORDER_COLOR_DISABLED: LINE_COLOR)),
+    GuiDrawRectangle(bounds, RAYGUI_PANEL_BORDER_WIDTH, GetColor(GuiGetStyle(DEFAULT, (state == STATE_DISABLED)? (int)BORDER_COLOR_DISABLED: (int)LINE_COLOR)),
                      GetColor(GuiGetStyle(DEFAULT, (state == STATE_DISABLED)? BASE_COLOR_DISABLED : BACKGROUND_COLOR)));
     //--------------------------------------------------------------------
 

